application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarksCommand.java

```diff
@@ -0,0 +1,268 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.emoji.Emoji;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.SubcommandData;
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.BookmarksRecord;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BookmarksCommand extends SlashCommandAdapter {
+
+    public static final long EXPIRE_DELAY_SECONDS = 3L * 24 * 60 * 60; // 3 Days
+
+
+    private static final MessageEmbed EMBED_NOT_IN_DM =
+            genSimpleEmbed("This command cannot be used in a DM!", 0xFF8484);
+
+    private static final MessageEmbed EMBED_NOT_IN_HELP_THREAD =
+            genSimpleEmbed("This command can only be used in help threads!", 0xFF8484);
+
+    private static final MessageEmbed EMBED_ALREADY_BOOKMARKED =
+            genSimpleEmbed("You have already bookmarked this thread!", 0xFF8484);
+
+    private static final MessageEmbed EMBED_BOOKMARK_REMOVED =
+            genSimpleEmbed("The bookmark was removed!", 0xFFA500);
+
+
+    private final Database database;
+    private static final Map<String, BookmarksPaginator> bookmarksPaginators = new HashMap<>();
```

Comment:
1: Ok gonna change that
2: To easily access the paginates, they are added by a UUID anyways
3: It is cleaned up if the paginator expired. I have never looked into caffeeine though so I will do that
4: Yes but that doesn't really matter as this will almost never happened and even if the impact is insignificant. I would appreciate a better approach to this though

Reply:
> what does this map do? a few things that seem odd to me (but i first need to understand what it does):